### ISSUE TITLE
[LES] `eth_getTransactionReceipt` returns null when transaction has been mined

### ISSUE THREAD

**Issue by rkeene** [ethereum/go-ethereum#16347]
geth v1.8.2 running in light mode on Linux/x86_64 synced with 4 peers

https://etherscan.io/tx/0x5d9a62b8a4aea33e86dc58abaec4dbd89e61eacbf4ab52fda332786f34fb0bf5

```
$ curl --header "Content-type: application/json" -sS -X POST --data '{"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0x5d9a62b8a4aea33e86dc58abaec4dbd89e61eacbf4ab52fda332786f34fb0bf5"],"id":1}' http://54.89.237.211/
{"jsonrpc":"2.0","id":1,"result":null}
$ 
```

```
$ geth attach
> eth.getTransactionReceipt("0x5d9a62b8a4aea33e86dc58abaec4dbd89e61eacbf4ab52fda332786f34fb0bf5")
null
> eth.syncing
false
> eth.blockNumber
5286470
```

Is it expected that light nodes cannot be used for verifying transactions ?

**Comment by rkeene** [comment]
`eth.getTransactionFromBlock()` in the JavaScript console succeeds:

```
> eth.getTransactionFromBlock(5243155, 32)
{
  blockHash: "0x3e7ce55799378a2b8ce47b319a224cec7d3445aaac3e45ac3fe2a23661a8d6ad",
  blockNumber: 5243155,
  from: "0x2b5634c42055806a59e9107ed44d43c426e58258",
  gas: 90000,
  gasPrice: 15000000000,
  hash: "0x5d9a62b8a4aea33e86dc58abaec4dbd89e61eacbf4ab52fda332786f34fb0bf5",
  input: "0x",
  nonce: 567361,
  r: "0x894a06cae04daad500ca26f1f4b3ba0d7533894b36d6648f12da4216c3a3446a",
  s: "0x6f183807332e3347f08ea9faa49a6469ecbe62eee2ec4ca73b018f603b3e0718",
  to: "0x2794a97c8cef516c044029ec4f799b7d24fab586",
  transactionIndex: 32,
  v: "0x26",
  value: 9990000000000000000
}
>
```

But `eth.getTransactionReceipt()` fails.

**Comment by dimitar-petrov** [comment]
I am facing same problem.
Did you find a solution?

**Comment by chuchuhao** [comment]
same problem ...

**Comment by sunyangf2** [comment]
seems all txs of block 6693401 have the problem.

**Comment by yqzzj** [comment]
same problem...
v1.8.27

**Comment by mw-cyzen** [comment]
is this still an issue

**Comment by karalabe** [comment]
This should have been fixed in 1.9.0 already. Please reopen if not.

**Comment by veox** [comment]
This must've resurfaced, I'm trying this in v1.9.10 and getting same buggy behaviour (in `console`).
